diff --git a/libavformat/libsmbclient.c b/libavformat/libsmbclient.c
index 28de2da..315b378 100644
--- a/libavformat/libsmbclient.c
+++ b/libavformat/libsmbclient.c
@@ -33,6 +33,8 @@ typedef struct {
     int trunc;
     int timeout;
     char *workgroup;
+    char *username;
+    char *password;
 } LIBSMBContext;
 
 static void libsmbc_get_auth_data(SMBCCTX *c, const char *server, const char *share,
