diff --git a/libavformat/libsmbclient.c b/libavformat/libsmbclient.c
index 28de2da..1aafefe 100644
--- a/libavformat/libsmbclient.c
+++ b/libavformat/libsmbclient.c
@@ -354,6 +354,8 @@
 static const AVOption options[] = {
+    {"username", "set the username used for SMB authentication", OFFSET(username), AV_OPT_TYPE_STRING, { 0 }, 0, 0, D|E },
+    {"password", "set the password used for SMB authentication", OFFSET(password), AV_OPT_TYPE_STRING, { 0 }, 0, 0, D|E },
     {"timeout",   "set timeout in ms of socket I/O operations",    OFFSET(timeout), AV_OPT_TYPE_INT, {.i64 = -1}, -1, INT_MAX, D|E },
     {"truncate",  "truncate existing files on write",              OFFSET(trunc),   AV_OPT_TYPE_INT, { .i64 = 1 }, 0, 1, E },
     {"workgroup", "set the workgroup used for making connections", OFFSET(workgroup), AV_OPT_TYPE_STRING, { 0 }, 0, 0, D|E },
     {NULL}
 };
