diff --git a/libavformat/libsmbclient.c b/libavformat/libsmbclient.c
index 28de2da..1aafefe 100644
--- a/libavformat/libsmbclient.c
+++ b/libavformat/libsmbclient.c
@@ -356,0 +357,2 @@ static const AVOption options[] = {
+    {"username", "set the username used for SMB authentication", OFFSET(username), AV_OPT_TYPE_STRING, { 0 }, 0, 0, D|E },
+    {"password", "set the password used for SMB authentication", OFFSET(password), AV_OPT_TYPE_STRING, { 0 }, 0, 0, D|E },
