diff --git a/scripts/mpv.sh b/scripts/mpv.sh
index 5bf12e1..54fae0e 100755
--- a/scripts/mpv.sh
+++ b/scripts/mpv.sh
@@ -23,15 +23,16 @@ cd .build
 meson setup .. \
+  --wipe \
   --cross-file $ROOT_DIR/libmpv/arm64-crossfile.ini \
   --prefix=$DEST/mpv \
   --default-library shared \
   --strip \
   -Dopensles=disabled \
   -Dohos=enabled \
   -Degl-ohos=enabled \
   -Dvulkan=enabled \
   -Dshaderc=enabled \
   -Dlua=enabled \
-  -Dgpl=false \
+  -Dgpl=true \
   -Dbuild-date=false \
   -Dcplayer=false \
   -Dmanpage-build=disabled
