--- a/libmpv/mpv/meson.build
+++ b/libmpv/mpv/meson.build
@@ -487,8 +487,7 @@
                      cc.find_library('native_image'),
                      cc.find_library('native_window')]
     sources += files('audio/out/ao_ohaudio.c',
-                     'video/out/ohos_common.c',
-                     'video/out/hwdec/hwdec_ohcodec.c')
+                     'video/out/ohos_common.c')
 endif
 
 uwp_opt = get_option('uwp').require(
@@ -1258,8 +1257,7 @@
 if features['egl-ohos']
     dependencies += egl_ohos
     features += {'gl': true}
-    sources += files('video/out/opengl/context_ohos.c',
-                     'video/out/hwdec/hwdec_ohcodec_gl.c')
+    sources += files('video/out/opengl/context_ohos.c')
 endif
 
 egl_drm = get_option('egl-drm').require(
@@ -1358,8 +1356,7 @@
 endif
 
 if features['vulkan'] and features['ohos']
-     sources += files('video/out/vulkan/context_ohos.c',
-                      'video/out/hwdec/hwdec_ohcodec_pl.c')
+     sources += files('video/out/vulkan/context_ohos.c')
 endif
 
 features += {'vk-khr-display': vulkan.type_name() == 'internal' or
--- a/libmpv/mpv/video/out/gpu/hwdec.c
+++ b/libmpv/mpv/video/out/gpu/hwdec.c
@@ -37,7 +37,6 @@
 extern const struct ra_hwdec_driver ra_hwdec_drmprime;
 extern const struct ra_hwdec_driver ra_hwdec_drmprime_overlay;
 extern const struct ra_hwdec_driver ra_hwdec_aimagereader;
-extern const struct ra_hwdec_driver ra_hwdec_ohcodec;
 extern const struct ra_hwdec_driver ra_hwdec_vulkan;
 
 const struct ra_hwdec_driver *const ra_hwdec_drivers[] = {
@@ -76,9 +75,6 @@
 #endif
 #if HAVE_ANDROID_MEDIA_NDK
     &ra_hwdec_aimagereader,
-#endif
-#if HAVE_OHOS
-    &ra_hwdec_ohcodec,
 #endif
 #if HAVE_VULKAN
     &ra_hwdec_vulkan,
